framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -362,14 +370,16 @@ public AnnotatedTypeMirror visitWildcard(WildcardTree node,
     @Override
     public AnnotatedTypeMirror visitPrimitiveType(PrimitiveTypeTree node,
                                                   AnnotatedTypeFactory f) {
-        // for e.g. "int.class"
+        // Node is the primitive type when primitive class literals are visited.
```

Comment:
Then aren't you afraid that this might introduce inconsistencies between the two?
The visitMemberSelect logic uses f.getAnnotatedType(elt); as the type for int.class.
Whereas dataflow will use f.fromTypeTree(node); as the type for int. What does Dataflow do with that type? Use it also for int.class?